cpp/tests/io/type_inference_test.cu

```diff
@@ -0,0 +1,67 @@
+/*
```

Comment:
can this be a .cpp file?

Reply:
Good point. I think this is feasible if I split the current `type_inference.cuh` into `type_inference.hpp` and `type_inference.cu`. Let's do this.

Reply:
Updates: the changes to make this work is not trivial since `type_inference.cuh` includes several other `.cuh` containing device functions.

Reply:
That's all right, then. Not a huge gain from this.